src/code/z_kaleido_manager.c

```diff
@@ -1,15 +1,104 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_manager/D_801DF9C0.s")
+#define KALEIDO_OVERLAY(name)                                                                                \
+    {                                                                                                        \
+        NULL, (u32)_ovl_##name##SegmentRomStart, (u32)_ovl_##name##SegmentRomEnd, _ovl_##name##SegmentStart, \
+            _ovl_##name##SegmentEnd, 0, #name,                                                               \
```

Comment:
```suggestion
        NULL, SEGMENT_ROM_START(ovl_##name), SEGMENT_ROM_END(ovl_##name), SEGMENT_START(ovl_##name), \
            SEGMENT_END(ovl_##name), 0, #name,                                                               \
```
I believe you can just use these SEGMENT macros. Take a look at `src/code/z_actor_dlftbls.c` for an example.